### PR TITLE
Add Open Graph Info to Docs HTML

### LINKED ISSUE #1722: Add Open Graph Info to Docs HTML
The current documentation does not contain any open graph information. This means that if someone shares a link to the docs via social media they will not get a nice image with the share. Open graph information should be added for better presentation during sharing.

For reference, here is the relevant code in Featuretools that adds this information:
https://github.com/alteryx/featuretools/blob/65025377fd1012d5ebd241602da79cd1926679c3/docs/source/templates/layout.html#L12-L25

The attached image can be used for the open graph image.
OpenSource_OpenGraph_1200x630px-evalml

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #1722.

Uploaded Open Graph photo to https://featurelabs-static.s3.amazonaws.com/OpenSource_OpenGraph_1200x630px-evalml.png to match where Featuretools' equivalent is :D

See the update preview here: https://www.opengraph.xyz/url/https:%2F%2Fevalml.alteryx.com%2Fen%2F1722_open_graph_info%2F/
🥳 

image

Note: I had to activate and set public this branch to view this. Will remove after this is merged!

### CODE REVIEW

**Review by freddyaboulton — APPROVED** [review]
Looks really cool @angela97lin !

**Review by chukarsten — APPROVED** [review]
Looks good!

File: docs/source/_templates/layout.html [hunk 1]
```diff
@@ -9,6 +9,24 @@
   }}();
 </script> -->
 
+<title>EvalML - Alteryx Open Source AutoML Library</title>
+<meta name="description" content="EvalML is an AutoML library that builds, optimizes, and evaluates machine learning pipelines using domain-specific objective functions. — EvalML 0.17.0 documentation">
```
Comment (bchen1116) [inline docs/source/_templates/layout.html:13]: Are we leaving the version at 0.17 or updating to 0.18?

**Review by bchen1116 — APPROVED** [review]
LGTM! Left one question

File: docs/source/_templates/layout.html (hunk 1 above)
Comment (angela97lin) [reply docs/source/_templates/layout.html:13]: Oh good catch! I'll just delete the version altogether so it doesn't have to be constantly updated :P

**Pull request merged**